ui-modules/app-inspector/app/views/main/inspect/summary/summary.controller.js

```diff
@@ -51,14 +51,14 @@ export function summaryController($scope, $state, $stateParams, $q, $http, brSna
     let observers = [];
 
     entityApi.entity(applicationId, entityId).then((response)=> {
-        vm.entity = response.data;
-        vm.name = response.data.name;
-        vm.error.entity = undefined;
-        observers.push(response.subscribe((response)=> {
+        let set = (response) => {
             vm.entity = response.data;
             vm.name = response.data.name;
             vm.error.entity = undefined;
-        }));
+            iconService.get(response.data, true).then(value => vm.iconUrl = value);
```

Comment:
Should set the `entity.iconUrl` instead of adding this into the scope, so the icon will be saved on the model which is propagated everywhere.